### DISCUSSION TITLE
expose useHistory to allow back button handling on android

### DISCUSSION THREAD

**Discussion by fieteboerner** [quasarframework/quasar#17501] [Ideas / Proposals]
Hi there, 

### Background
i'am using your framework to build a mobile app for ios/android.
In this app i'am kind of stuck at the following problems
   1. i dont want to close a dialog, when the route changes (because i'am changing query-params with inputs inside this dialog)
   2. i need another representation of a dialog (drawer right below an input) which i want to close with the back button

I figured out quasar is handling the back button on android if a dialog is open. So i've diged into the codebase how it is handled. So i found the History.js and use-history.js. And i quite  like the approach to add something to this history so it is possible to hook in with the back button. 
But unfortunately, this API is not exposed at all and can only used inside of the library. I can imagine it would be also really helpful for other poeple to use this api, to allow to control custom elements/components using the back button in android.

### TL;DR / actual Question
Would you be open to merge a PR, where this useHistory- Helper would be exported to userland?
If yes I would, what would be requirements you would merge this. I would like to contribute a PR. :)

Thanks!
